### DISCUSSION TITLE
File-system based router?

### DISCUSSION THREAD

**Discussion by ifsheldon** [DioxusLabs/dioxus#3224] [Ideas]
Hi all! I am a frontend newbie, learning general frontend technology and Dioxus. I find the file-system based router of Next.js map quite intuitively into my mental model, despite some concerns about dark framework knowledge not shown in the source code. 

Now Dioxus 0.5 and as far as I know the upcoming 0.6 still require us to define routes in code. So I wonder whether there is ongoing work on a file-system based router, and why you chose code-based router and why not file-system based router?

**Comment by ealmloff** [comment]
There was some discussion of file based routing in #922
